File: tts-pronunciations/js/example_tts_with_workspace_dictionary.js
#!/usr/bin/env node
/** Compare speech with the workspace's existing Portal pronunciation entries. */

const fs = require('node:fs/promises');
const path = require('node:path');
const os = require('node:os');
const { clientFromEnvironment } = require('./example_pronunciation_dictionaries');
const { decodeAudio } = require('./example_tts_with_dictionary');

async function runWorkspaceTts(client, {
    text = 'The Cat is on the mat.', io = fs, tempRoot = os.tmpdir(), log = console.log,
} = {}) {
    if (!text.trim()) throw new Error('Synthesis text must not be empty.');
    const request = {
        text, voiceId: 'Ashley', modelId: 'inworld-tts-2', language: 'en-US',
        audioConfig: { audioEncoding: 'MP3' }, seed: 101,
    };
    log('Two billable TTS requests; uses existing Portal entries without changing them.');
    const baseline = decodeAudio(await client.synthesize(request));
    const selected = decodeAudio(await client.synthesize({
        ...request, enable_custom_pronunciation: true,
    }));
    const directory = await io.mkdtemp(path.join(tempRoot, 'workspace-pronunciation-'));
    const baselinePath = path.join(directory, 'baseline.mp3');
    const selectedPath = path.join(directory, 'with-workspace-dictionary.mp3');
    await io.writeFile(baselinePath, baseline);
    await io.writeFile(selectedPath, selected);
    log(`Baseline audio: ${baselinePath}`);
    log(`Workspace-dictionary audio: ${selectedPath}`);
    log('If Portal has Cat → /ɹɛd/ for en-US, listen for cat → red in the second file.');
    log('HTTP success is not proof of rewriting: unavailable defaults leave speech unchanged.');
    return { baselinePath, selectedPath };
}

if (require.main === module) {
    Promise.resolve().then(() => runWorkspaceTts(clientFromEnvironment())).catch((error) => {
        console.error(`Workspace pronunciation example failed: ${error.message}`);
        process.exitCode = 1;
    });
}

module.exports = { runWorkspaceTts };

File: pronunciations/python/example_tts_with_workspace_dictionary.py
#!/usr/bin/env python3
"""Compare speech with existing Portal entries; never create or edit a dictionary."""

import sys
import tempfile
from pathlib import Path

from example_pronunciation_dictionaries import client_from_environment
from example_tts_with_dictionary import decode_audio


def synthesize_workspace_comparison(client, text="The Cat is on the mat."):
    if not text.strip():
        raise ValueError("Synthesis text must not be empty.")
    request = {
        "text": text,
        "voiceId": "Ashley",
        "modelId": "inworld-tts-2",
        "language": "en-US",
        "audioConfig": {"audioEncoding": "MP3"},
        "seed": 101,
    }
    print("Two billable TTS requests; uses existing Portal entries without changing them.")
    endpoint = f"{client.api_base_url}/tts/v1/voice"
    baseline = decode_audio(client._request("POST", endpoint, json=request))
    selected = decode_audio(client._request(
        "POST", endpoint, json={**request, "enable_custom_pronunciation": True}
    ))
    directory = Path(tempfile.mkdtemp(prefix="workspace-pronunciation-"))
    baseline_path = directory / "baseline.mp3"
    selected_path = directory / "with-workspace-dictionary.mp3"
    baseline_path.write_bytes(baseline)
    selected_path.write_bytes(selected)
    print(f"Baseline audio: {baseline_path}")
    print(f"Workspace-dictionary audio: {selected_path}")
    print("If Portal has Cat → /ɹɛd/ for en-US, listen for cat → red in the second file.")
    print("HTTP success is not proof of rewriting: unavailable defaults leave speech unchanged.")
    return directory


def main():
    try:
        with client_from_environment() as client:
            synthesize_workspace_comparison(client)
    except (OSError, RuntimeError, ValueError, KeyError) as error:
        print(f"Workspace pronunciation example failed: {error}", file=sys.stderr)
        return 1
    return 0


if __name__ == "__main__":
    sys.exit(main())
